ext/src/zpages/tracez_data_aggregator.cc

```diff
@@ -0,0 +1,110 @@
+  {
+    if(kLatencyBoundaries[boundary] <= recordable_duration && recordable_duration < kLatencyBoundaries[boundary+1])return (LatencyBoundaryName)boundary;
+  }
+  return LatencyBoundaryName::k100SecondToMax;
+}
+
+void TracezDataAggregator::AggregateStatusOKSpans(std::unique_ptr<opentelemetry::sdk::trace::Recordable>& ok_span)
+{
+  LatencyBoundaryName boundary_name = GetLatencyBoundary(ok_span.get());
+  std::string span_name = ok_span.get()->GetName().data();
+  
+  //If the sample span storage is at capacity, remove the span that was inserted earliest and free up memory
+  if(aggregated_data_[span_name].get()->latency_sample_spans_[boundary_name].size() == kMaxNumberOfSampleSpans)
+  {
+    aggregated_data_[span_name].get()->latency_sample_spans_[boundary_name].front().reset();
+    aggregated_data_[span_name].get()->latency_sample_spans_[boundary_name].pop_front();
+  }
+  aggregated_data_[span_name].get()->latency_sample_spans_[boundary_name].push_back(std::move(ok_span));
+  aggregated_data_[span_name].get()->span_count_per_latency_bucket_[boundary_name]++;
+}
+
+void TracezDataAggregator::AggregateStatusErrorSpans(std::unique_ptr<opentelemetry::sdk::trace::Recordable>& error_span)
+{
+  std::string span_name = error_span.get()->GetName().data();
+  
+  //If the error samples storage is at capacity, remove the span that was inserted earliest and free up memory
+  if(aggregated_data_[span_name].get()->error_sample_spans_.size() == kMaxNumberOfSampleSpans)
+  {
+    aggregated_data_[span_name].get()->error_sample_spans_.front().reset();
+    aggregated_data_[span_name].get()->error_sample_spans_.pop_front();
+  }
+  aggregated_data_[span_name].get()->error_sample_spans_.push_back(std::move(error_span));
+  aggregated_data_[span_name].get()->error_spans_++;
+}
+
+void TracezDataAggregator::AggregateCompletedSpans()
+{
+  auto completed_spans = tracez_span_processor_->GetCompletedSpans();
+  
+  for(auto& span: completed_spans)
+  {
+    std::string span_name = span.get()->GetName().data();
+    
+    if(aggregated_data_.find(span_name) == aggregated_data_.end())
+    aggregated_data_[span_name] = std::unique_ptr<AggregatedInformation>(new AggregatedInformation);
+    
+    //running spans are calculated from scratch later
+    aggregated_data_[span_name].get() -> running_spans_ = 0;
+    
+    if(span.get()->GetStatus() == opentelemetry::trace::CanonicalCode::OK)AggregateStatusOKSpans(span);
+    else AggregateStatusErrorSpans(span);
+  }
+}
+
+void TracezDataAggregator::AggregateRunningSpans()
+{
+  auto running_spans = tracez_span_processor_->GetRunningSpans();
+  std::unordered_set<std::string> cache;
+  for(auto running_span: running_spans)
+  {
+    std::string span_name = running_span->GetName().data();
+ 
+    if(aggregated_data_.find(span_name) == aggregated_data_.end())
+    aggregated_data_[span_name] = std::unique_ptr<AggregatedInformation>(new AggregatedInformation);
```

Comment:
Nit: Indentation/parentheses will simplify reading the code. 

Also no need to have 'new AggregatedInformation', instead:
aggregated_data_[span_name] = std::make_unique<AggregatedInformation>();

Reply:
I tried this initially it gives me an error saying make_unique is not a member of std and I found a link to this answer (https://stackoverflow.com/questions/24609271/errormake-unique-is-not-a-member-of-std)

Reply:
Oh, I see. There is also absl::make_unique, but not sure if we can use absl in the repo.